pkg/client/src/app/pages/applications/analysis-wizard/components/add-custom-rules.tsx

```diff
@@ -17,114 +16,178 @@ import { IReadFile } from "../analysis-wizard";
 import { useFormContext } from "react-hook-form";
 
 const xmllint = require("xmllint");
-interface IAddCustomRulesProps {
-  currentFiles: IReadFile[];
-  setCurrentFiles: (files: IReadFile[]) => void;
-}
-
-export const AddCustomRules: React.FunctionComponent<IAddCustomRulesProps> = ({
-  currentFiles,
-  setCurrentFiles,
-}) => {
+
+export const AddCustomRules: React.FunctionComponent = () => {
   const { getValues, setValue } = useFormContext();
-  const customRulesFiles: IReadFile[] = getValues("customRulesFiles");
-  const [modalText, setModalText] = React.useState("");
-  const [isLoading, setIsLoading] = React.useState(false);
+  const [readFileData, setReadFileData] = React.useState<IReadFile[]>([]);
   const [error, setError] = React.useState("");
+  const [currentFiles, setCurrentFiles] = React.useState<File[]>([]);
+  const [showStatus, setShowStatus] = React.useState(false);
+  const [statusIcon, setStatusIcon] = React.useState("inProgress");
+
+  // only show the status component once a file has been uploaded, but keep the status list component itself even if all files are removed
+  if (!showStatus && currentFiles.length > 0) {
+    setShowStatus(true);
+  }
 
-  const getStatusIcon = () => {
-    if (isLoading) {
-      return "inProgress";
-    } else if (currentFiles.every((file) => file.loadResult === "success")) {
-      return "success";
+  // determine the icon that should be shown for the overall status list
+  React.useEffect(() => {
```

Comment:
@gildub it looks like you're reversing a lot of the changes from https://github.com/konveyor/tackle2-ui/pull/264, such as restoring these pieces of redundant/duplicated state and logic that was moved from effects to render-time. Is there a way we can address the issue without losing those code quality improvements? This feels like a bit of a step backwards. At a minimum we definitely shouldn't reintroduce state that is only updated from changes in props (they can simply be plain variables).

If it's because there were a lot of conflicts, we could look at resolving them more cleanly together if you're up for that? I wonder if we can look at a diff of your changes before #264 (do you have those in your reflog from before a rebase, maybe?) to see how we can reapply them to the new code more surgically.

cc @ibolton336

Reply:
Of course, I'll organize a meeting. Thanks

Reply:
@mturley, regarding this `useEffect()`, it actually comes from latest PF for file upload multiple (and still moving being in beta):
https://codesandbox.io/s/nvslhe.

Reply:
@gildub that's interesting, thanks for pointing that out. I think that is a flaw in that PF example to be honest. I'll make a note to follow up with them about that, it's definitely an antipattern. If our code works without it we should remove it IMO.

Reply:
@mturley, ok thanks for feedback, I'll look into it.